map.go

```diff
@@ -20,6 +20,18 @@ var (
 	ErrIterationAborted = errors.New("iteration aborted")
 )
 
+type ErrMapIncompatible struct {
+	mapName string
+}
+
+func NewErrMapIncompatible(mapName string) ErrMapIncompatible {
+	return ErrMapIncompatible{mapName}
+}
+
+func (e ErrMapIncompatible) Error() string {
+	return fmt.Sprintf("incompatible map's spec for pinned map %s", e.mapName)
```

Comment:
I slightly changed the error message to try make it more explicit (c.f. https://github.com/cilium/cilium/issues/15629)